packages/changed-elements-react/src/widgets/V2Widget/hooks/useNamedVersionLoader.tsx

```diff
@@ -0,0 +1,354 @@
+      description: currentChangeSet.description,
+      createdDateTime: currentChangeSet.pushDateTime,
+    };
+  }
+  return;
+};
+
+const sortNamedVersions = (namedVersions: NamedVersion[], currentNamedVersion: NamedVersion, onError: () => void) => {
+  //if current index is 0 then no need to filter. All change sets are older than current.
+  const namedVersionsOlderThanCurrentVersion = currentNamedVersion.changesetIndex !== 0 ? namedVersions.filter(version => version.changesetIndex <= currentNamedVersion.changesetIndex) :
+    namedVersions;
+  if (namedVersionsOlderThanCurrentVersion.length === 0) {
+    onError();
+    return;
+  }
+  return namedVersionsOlderThanCurrentVersion.reverse();
+};
+
+type ProcessChangesetsArgs = {
+  iTwinId: string;
+  iModelId: string;
+  namedVersionLoaderState: NamedVersionLoaderState;
+  comparisonJobClient: IComparisonJobClient;
+  setResult: (result: NamedVersionLoaderResult) => void;
+  isDisposed: () => boolean;
+};
+
+type Entry = {
+  version: NamedVersion;
+  state: VersionProcessedState;
+  jobStatus: JobStatus;
+  jobProgress: JobProgress;
+};
+
+const processChangesetsAndUpdateResultState = async (args: ProcessChangesetsArgs) => {
+  const currentVersionId = args.namedVersionLoaderState.result.namedVersions.currentVersion.version.id;
+
+  // deletes all comparison jobs only for debugging purposes
+  // await Promise.all(args.namedVersionLoaderState.result.namedVersions.entries.map(async (entry) => {
+  //await deleteJob(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+  // })).catch(() => { return; });
+  const newEntries = await Promise.all(args.namedVersionLoaderState.result.namedVersions.entries.map(async (entry) => {
+    const jobStatusAndJobProgress: JobStatusAndJobProgress = await getJobStatusAndJobProgress(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+    return {
+      version: entry.version,
+      state: VersionProcessedState.Processed,
+      jobStatus: jobStatusAndJobProgress.jobStatus,
+      jobProgress: jobStatusAndJobProgress.jobProgress,
+    };
+  }));
+  args.namedVersionLoaderState.result = {
+    namedVersions: { currentVersion: args.namedVersionLoaderState.result.namedVersions.currentVersion, entries: newEntries },
+  };
+  args.setResult(args.namedVersionLoaderState.result);
+  void pollForProgressingJobs({
+    iTwinId: args.iTwinId,
+    iModelId: args.iModelId,
+    namedVersionLoaderState: args.namedVersionLoaderState,
+    comparisonJobClient: args.comparisonJobClient,
+    setResult: args.setResult,
+    isDisposed: args.isDisposed,
+  });
+};
+
+type UpdateProgressingChangesetsArgs = {
+  iTwinId: string;
+  iModelId: string;
+  namedVersionLoaderState: NamedVersionLoaderState;
+  comparisonJobClient: IComparisonJobClient;
+  setResult: (result: NamedVersionLoaderResult) => void;
+  isDisposed: () => boolean;
+};
+const pollForProgressingJobs = async (args: UpdateProgressingChangesetsArgs) => {
```

Comment:
Wording suggestion: Use `InProgress` instead of `Progressing`.

Not sure if there's more, but right around this code I see you could update the names of pollForProgressingJobs , UpdateProgressingChangesetsArgs, and areThereProgressingJob

Reply:
I see you took my suggestion for the function name. Can you similarly update the name of `UpdateProgressingChangesetsArgs` and `areThereProgressingJob`?